EndlessRunnerTutorial/Assets/Scripts/PlayerMovement.cs

```diff
@@ -1,38 +1,90 @@
-﻿using UnityEngine;
+﻿using System;
+using UnityEngine;
 using UnityEngine.SceneManagement;
 
 public class PlayerMovement : MonoBehaviour {
 
-    bool alive = true;
 
     public float speed = 5;
-    [SerializeField] Rigidbody rb;
-
-    float horizontalInput;
     [SerializeField] float horizontalMultiplier = 2;
-
     public float speedIncreasePerPoint = 0.1f;
 
-    private void FixedUpdate ()
-    {
-        if (!alive) return;
+    private float horizontalInput;
+    private bool m_alive = true;
+    private Vector2 m_currentLane;
+    private Vector3 m_currentVel;
+    public float rotSpeed = 5f;
 
-        Vector3 forwardMove = transform.forward * speed * Time.fixedDeltaTime;
-        Vector3 horizontalMove = transform.right * horizontalInput * speed * Time.fixedDeltaTime * horizontalMultiplier;
-        rb.MovePosition(rb.position + forwardMove + horizontalMove);
+    private void Start()
+    {
+        m_currentLane = new Vector2(2.5f, 0);
     }
 
-    private void Update () {
-        horizontalInput = Input.GetAxis("Horizontal");
 
-        if (transform.position.y < -5) {
-            Die();
+    private void Update()
+    {
+        if (transform.position.y < -5) Die();
+        if (!m_alive) return;
+
+        // Forward Movement
+        Vector3 forwardMove = Vector3.forward * speed * Time.fixedDeltaTime;
```

Comment:
@BhavsarJay  this needs to be `Time.deltaTime`
```suggestion
        Vector3 forwardMove = Vector3.forward * speed * Time.deltaTime;
```